### PR TITLE
Encode illegal characters inside cookies and change cookies parsing logic

### PR DESCRIPTION
**Pull request by borisfba**
*Issue # (if available):*
#43 

*Description of changes:*
* Make produced cookies RFC 6265 compliant by URI encoding illegal characters.
* Revise cookies parsing logic to fix issues with subdomains (#43)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jeandek — APPROVED** [review]
LGTM

File: src/index.ts [hunk 1]
```diff
@@ -175,24 +162,25 @@ export class Authenticator {
...
+    this._logger.debug({ msg: 'Extracting authentication token from request cookie', cookieHeaders });
+
+    const tokenCookieNamePrefix = `${this._cookieBase}.`;
+    const tokenCookieNamePostfix = '.idToken';
+
+    const cookies = cookieHeaders.flatMap(h => Cookies.parse(h.value));
+    const token = cookies.find(c => c.name.startsWith(tokenCookieNamePrefix) && c.name.endsWith(tokenCookieNamePostfix))?.value;
+
+    if (!token) {
+      this._logger.debug("  idToken wasn't present in request cookies");
```
Comment (ckifer) [inline src/index.ts:178]: just tiny nit: why is there a space in this log (ik it was there before)

File: src/index.ts [hunk 2]
```diff
@@ -175,24 +162,25 @@ export class Authenticator {
...
+      throw new Error("Id token isn't present in the request cookies");
```
Comment (ckifer) [inline src/index.ts:179]: also tiny nit that was here before: consistency with "idToken" above or "Id token" here

File: src/index.ts [hunk 3]
```diff
@@ -175,24 +162,25 @@ export class Authenticator {
...
     }
-    this._logger.debug("  idToken wasn't present in request cookies");
-    throw new Error("Id token isn't present in the request cookies");
+
+    this._logger.debug({ msg: '  Found token in cookie', token });
```
Comment (ckifer) [inline src/index.ts:182]: nit here before: space in log

**Review by ckifer — APPROVED** [review]
Awesome stuff, thanks so much for doing this!

Should also close my draft PR #44

File: src/index.ts (hunk 1 above)
Comment (borisfba) [reply src/index.ts]: True, space shouldn't be here. Thanks for pointing out!

**Pull request merged**